src/main/scala/InsufficientMatingMaterial.scala

```diff
@@ -8,54 +8,86 @@ package chess
  */
 object InsufficientMatingMaterial {
 
-  def nonKingPieces(board: Board) = board.pieces.filter(_._2.role != King).toList
+  def nonKingPieceMap(board: Board) = board.pieces.filter {case (_, piece) => piece isNot King}
+  def nonKingNonBishopPieceMap(board: Board) = board.pieces.filter {case (_, piece) => (piece isNot King) && (piece isNot Bishop)}
+  def nonKingPieces(board: Board) = nonKingPieceMap(board).toList
+  def boardWithNonKingPieces(board: Board) = board withPieces nonKingPieceMap(board)
+  def boardWithNonKingNonBishopPieces(board: Board) = board withPieces nonKingNonBishopPieceMap(board)
 
   /**
-   * Returns true when the only non-king pieces that remain are bishops that cannot
-   * capture each other and cannot checkmate (in atomic chess).
+   * Returns true when only the bishops are mobile and the bishops cannot capture.
+   * Assumes bishops can move through friendly pieces.
    */
-  def bishopsOnDifferentColor(board: Board) = {
+  def bishopsCannotCapture(board: Board) = noPieceMoves(boardWithNonKingNonBishopPieces(board)) && {
```

Comment:
Perhaps bishops with legal moves should be treated differently than immobile bishops.

For example, `2b5/6p1/3K2pb/6p1/6p1/3k2P1/8/8 w - -` has an immobile bishop that prevents the capture check from triggering draw, and `8/8/4B3/p1p5/P1P1k3/PBp5/P1P5/N3K3 w - -` has an immobile bishop that prevents the no-non-bishop-moves from succeeding because the check removes immobile bishops before checking mobility.